plugins/modules/ndo_vlan_pool.py

```diff
@@ -0,0 +1,246 @@
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a new vlan pool
+  cisco.mso.ndo_vlan_pool:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    vlan_pool: ansible_test_vlan_pool
+    vlan_ranges:
+      - from: 100
+        to: 200
+    state: present
+
+- name: Query a vlan pool with template_name
+  cisco.mso.ndo_vlan_pool:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    vlan_pool: ansible_test_vlan_pool
+    state: query
+  register: query_one
+
+- name: Query all vlan pools in the template
+  cisco.mso.ndo_vlan_pool:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    state: query
+  register: query_all
+
+- name: Delete a vlan pool
+  cisco.mso.ndo_vlan_pool:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    vlan_pool: ansible_test_vlan_pool
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate, KVPair
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True),
+        vlan_pool=dict(type="str", aliases=["name"]),
+        vlan_pool_uuid=dict(type="str", aliases=["uuid"]),
+        description=dict(type="str"),
+        vlan_ranges=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                from_vlan=dict(type="int", required=True),
+                to_vlan=dict(type="int", required=True),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["vlan_pool"]],
+            ["state", "present", ["vlan_pool"]],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    vlan_pool = module.params.get("vlan_pool")
+    vlan_pool_uuid = module.params.get("vlan_pool_uuid")
+    vlan_ranges = get_vlan_ranges_payload(module.params.get("vlan_ranges")) if module.params.get("vlan_ranges") else []
+    description = module.params.get("description")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+    err_message_min_vlan_ranges = "At least one vlan range is required when state is present."
```

Comment:
Can you move it under the state == "present" check?

Reply:
moved